### PR TITLE
Add vital status to eCR viewer

### LINKED ISSUE #1342: Add Vital Status to eCR Viewer
## Action Requested
Add a field for Vital Status to the eCR Viewer.
> eCR Viewer
>> Patient Info
>>> Demographics
>>>> Vital Status

## Acceptance Criteria
Vital Status is displayed for all eCRs that have a vital status field.

## User Value
For certain conditions, it's useful or required by the CDC to have the patient's Vital Status. The Vital Status should either be "Alive" or "Deceased"

## Testing
(Sarah does not know this info)

## Designs
[Figma Designs](https://www.figma.com/file/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?type=design&node-id=1024%3A136062&mode=design&t=lClwL7CnvrviunUF-1)
[Figma Clickable Prototype](https://www.figma.com/proto/tXWUeV3Kjehd7VNqHojZYq/eCR-Viewer?page-id=871%3A2283&type=design&node-id=1606-27619&viewport=1011%2C558%2C0.05&t=sGUJKIN0MXVFYIfm-1&scaling=min-zoom&starting-point-node-id=1606%3A27619&mode=design)

## Design Review
Things to look for:
- check field name and capitalization match design
- check for ordering (should be below current age, or DOB if current age ticket hasn't been picked up yet)

## Background & Context
Not sure if this is currently converted

### PR DESCRIPTION
**Pull request by angelathe**
# PULL REQUEST

## Summary
- Add vital status to eCR viewer (& FHIR conversion)

Misc:
- Update seed data
- Update Demographics snapshot test

## Related Issue
Fixes #1342 

## Screenshot
Screenshot 2024-03-29 at 13 49 33

## References
- [Patient Deceased](https://cdasearch.hl7.org/examples/view/4f17c14a-a779-49aa-9b3d-b8f836ea8958)
- [CDA sdtc Extensions](https://confluence.hl7.org/display/SD/CDA+Extensions)
- [FHIR Resource Patient](https://build.fhir.org/patient.html)

## Checklist

- [ ] Vital Status should either be "Alive" or "Deceased"
- [ ] `deceasedBoolean` should be type `boolean`

## Design Review
- [ ] check field name and capitalization match design
- [ ] check for ordering (should be below current age, or DOB if current age ticket hasn't been picked up yet)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

File: containers/ecr-viewer/seed-scripts/fhir_data/09f39529-69b8-4937-8e01-596b7e966d87.json [hunk 1]
```diff
@@ -2323,18 +2348,6 @@
...
-            "low": {
-              "value": 36,
-              "unit": "%"
-            },
-            "high": {
-              "value": 43,
-              "unit": "%"
-            }
-          }
-        ],
```
Comment (BobanL) [inline containers/ecr-viewer/seed-scripts/fhir_data/09f39529-69b8-4937-8e01-596b7e966d87.json:2337]: ooo 👀 This is a bug. I'm going to open a PR to fix this.

File: containers/ecr-viewer/src/app/utils.tsx [hunk 2]
```diff
@@ -259,6 +259,12 @@ export const evaluateDemographicsData = (
       title: "Current Age",
       value: calculatePatientAge(fhirBundle, mappings)?.toString(),
     },
+    {
+      title: "Vital Status",
+      value: evaluate(fhirBundle, mappings.patientVitalStatus)[0]
+        ? "Deceased"
+        : "Alive",
```
Comment (BobanL) [inline containers/ecr-viewer/src/app/utils.tsx:266]: I'm going to assume that if that's the correct logic - If they aren't for sure Deceased then we can default them to being alive.

**Review by BobanL — APPROVED** [review]
Code looks good!

File: containers/ecr-viewer/src/app/utils.tsx (hunk 2 above)
Comment (angelathe) [reply containers/ecr-viewer/src/app/utils.tsx:266]: @maryyeh Does this logic sound alright to you?

**Comment by maryyeh** [comment]
@angelathe  - Vital status looks right. I think we could add "Deceased, [date]" if we wanted to be fancy, but I would defer that decision to Sarah as an enhancement, if we wanted. So, for now -- it's good!

**Pull request merged**